pkg/build/linter.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Chainguard, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build
+
+import (
+	"fmt"
+	"io/fs"
+	"regexp"
+
+	"chainguard.dev/melange/pkg/config"
+)
+
+type LinterContext struct {
+	pkgname string
+	cfg     *config.Configuration
+	chk     *config.Checks
+}
+
+type LinterFunc func(lctx LinterContext, path string, d fs.DirEntry) error
+
+type Linter struct {
+	LinterFunc LinterFunc
+	Explain    string
+}
+
+var Linters = map[string]Linter{
+	"usrlocal":  Linter{usrLocalLinter, "This package should be a -compat package"},
+	"varempty":  Linter{varEmptyLinter, "Remove any offending files in /var/empty in the pipeline"},
```

Comment:
The regex seems to look for files matching `/var/local` at all, and not be limited to `/var/empty` -- is the comment correct?